src/lib/fpti.js

```diff
@@ -0,0 +1,134 @@
+/* @flow */
+import { getClientID, getMerchantID, getPartnerAttributionID } from '@paypal/sdk-client/src';
+
+import { getDeviceInfo } from './get-device-info';
+
+const sendBeacon = (src, data) => {
+    if (!src || !data) {
+        return;
+    }
+
+    let query = Object.keys(data).map(key => {
+        return `${ encodeURIComponent(key) }=${ encodeURIComponent(data[key]) }`;
+    });
+
+    query = query.join('&');
+
+    const beaconImage = new window.Image();
+    beaconImage.src = `${ src }?${ query }`;
+};
+
+// removes empty strings, `undefined`, `null`, and `NaN` from fpti event
+const filterFalsyValues = source => {
```

Comment:
Trying to think.. maybe we'd actually want to know on the serverside if a value like undefined or NaN was passed. Could help w/ debugging, etc. What do you think?

Reply:
Ah, I'd rather explicitly log error conditions when we find them rather than allowing them to clutter here. Also we could technically identify a problem just by the fact certain values would be missing.